ckanext/harvest/logic/action/update.py

```diff
@@ -559,6 +564,116 @@ def harvest_jobs_run(context, data_dict):
+        return False
+
+    return True
+
+
+def send_error_mail(context, job_obj):
+    model = context['model']
+
+    for row in model.Session\
+            .query(model.Package)\
+            .filter(model.Package.id == job_obj.source.id):
+        pkg = row.as_dict()
+        harvest_name = str(pkg['name'])
+
+    ckan_site_url = config.get('ckan.site_url')
+    job_url = ckan_site_url + '/harvest/' + harvest_name + '/job/' + job_obj.id
+
+    msg = toolkit._('This is a failure-notification of the latest harvest job ({0}) set-up in {1}.').format(job_url, ckan_site_url)
+    msg += '\n\n'
+
+    for org, job in model.Session\
+            .query(model.Group, HarvestSource)\
+            .outerjoin(model.Member, model.Group) \
+            .join(HarvestSource, HarvestSource.id == model.Member.table_id) \
+            .filter(model.Member.table_id == job_obj.source.id):
+
+        if org:
+            org_dict = org.as_dict()
+            msg += toolkit._('Organization: {0}').format(org_dict['title'])
+            msg += '\n\n'
+
+        job_dict = job.as_dict()
+        msg += toolkit._('Harvest Job Title: {0}').format(job_dict['title'])
+        msg += '\n\n'
+
+    msg += toolkit._('Date of Harvest: {0}').format(str(job_obj.created))
+    msg += ' UTC\n\n'
+
+    out = {
+        'last_job': None,
+    }
+
+    out['last_job'] = harvest_job_dictize(job_obj, context)
+
+    msg += toolkit._('Records in Error: {0}').format(str(out['last_job']['stats'].get('errored', 0)))
+    msg += '\n'
+
+    obj_error = ''
+    job_error = ''
+
+
+    for harvest_object_error in model.Session.query(HarvestObjectError) \
+            .join(HarvestObject, HarvestObjectError.harvest_object_id == HarvestObject.harvest_job_id) \
+            .filter(HarvestObject.harvest_job_id == job_obj.id).limit(20).all():
+        error_dict = harvest_object_error.as_dict()
+        obj_error += error_dict['msg'] + '\n'
+
+    for harvest_gather_error in model.Session.query(HarvestGatherError) \
+        .filter(HarvestGatherError.harvest_job_id == job_obj.id):
+        harvest_gather_error_dict = harvest_gather_error.as_dict()
+        job_error += harvest_gather_error_dict['message'] + '\n'
+
+    if (obj_error != '' or job_error != ''):
+        msg += toolkit._('Error Summary')
+        msg += '\n\n'
+
+    if (obj_error != ''):
+        msg += toolkit._('Document Error')
+        msg += '\n' + obj_error + '\n\n'
+
+    if (job_error != ''):
+        msg += toolkit._('Job Errors')
+        msg += '\n' + job_error + '\n\n'
+
+    if obj_error or job_error:
+        msg += '\n--\n'
+        msg += toolkit._('You are receiving this email because you are currently set-up as Administrator for {0}. Please do not reply to this email as it was sent from a non-monitored address.').format(config.get('ckan.site_title'))
+
+        # get recipients
+        email_recipients = config.get('email_to').split(',')
```

Comment:
If the goal is to send it to all sysadmin users it is much safer to query them from the database as I suspect `email_to` is not always filled:

```python
sysadmins = model.Session.query(model.User).filter(model.User.sysadmin==True).all()

for user in sysadmins:
   
   # Use user.email


```